examples/custom_generic_nl_family_custom_types.rs

```diff
@@ -0,0 +1,138 @@
+        U32Bitmask::empty(),
+    )
+    .unwrap();
+    // family_name to family_id (temporarily assigned netlink family Id by kernel, after registering the generic netlink family)
+    let family_id = sock
+        .resolve_genl_family(family_name)
+        .expect("Kernel Module that registered custom family must exist!");
+    println!(
+        "Netlink family id of generic netlink family with name '{}' is: {}",
+        family_name, family_id
+    );
+
+    // We want to send an Echo command
+    // 1) prepare Foobar Attribute
+    let mut attrs: GenlBuffer<FoobarAttribute, Buffer> = GenlBuffer::new();
+    attrs.push(
+        Nlattr::new(
+            None,
+            false,
+            false,
+            FoobarAttribute::Msg,
+            "Some data that has `Nl` trait implemented, like &str",
+        )
+        .unwrap(),
+    );
+    // 2) prepare Generic Netlink Header. The Generic Netlink Header contains the
+    //    attributes (actual data) as payload.
+    let gnmsghdr = Genlmsghdr::new(
+        FoobarOperation::Echo,
+        // Version is dependent on your kernel module
+        // you can check it there and make special action, or you ignore it
+        1,
+        attrs,
+    );
+    // 3) Prepare Netlink header. The Netlink header contains the Generic Netlink header
+    //    as payload.
+    let nlmsghdr = Nlmsghdr::new(
+        None,
+        family_id,
+        // This depends on your kernel module. Do you check there if any flags are used?
+        // Request is required (TODO by neli or by netlink?)
+        // others are up to you
+        NlmFFlags::new(&[NlmF::Request]),
+        None,
+        // Port ID. Not necessarily the process id of the current process. This is not
+        // used for routing/addressing a socket. You can use this in a application-specific
+        // manner, e.g. to identify a specific socket if your process has multiple sockets.
+        Some(process::id()),
+        NlPayload::Payload(gnmsghdr),
+    );
+
+    // Send data
+    sock.send(nlmsghdr).expect("Send must work");
+
+    // receive echo'ed message
+    let res: Nlmsghdr<u16, Genlmsghdr<FoobarOperation, FoobarAttribute>> =
+        sock.recv().expect("Should receive a message").unwrap();
+
+    // According to Netlink Linux kernel code nl_type is either 0x2 (error) or our family id (ok)
+    if res.nl_type == u16::from(Nlmsg::Error)
+    /* Nlmsg::Error <==> NLMSG_ERR is 0x2, see Linux kernel source code */
+    {
+        // it actually depends on the kernel module if it replies with this nl_type
+        // depends on your specific use case
+        panic!("Kernel replied with error");
+    }
+    if res.nl_type == family_id {
+        println!("Received successful reply!");
+    }
+
+    let attr_handle = res.get_payload().unwrap().get_attr_handle();
+    let received = attr_handle.get_attribute(FoobarAttribute::Msg).unwrap();
+    let received = String::deserialize(received.nla_payload.as_ref()).unwrap();
```

Comment:
This can be more concisely represented by `attr_handle.get_attr_payload_as::<String>(FoobarAttribute::Msg)`.